src/library/relations/dlgrelations.cpp

```diff
@@ -0,0 +1,137 @@
+#include "library/relations/dlgrelations.h"
+
+#include <QItemSelection>
+
+#include "controllers/keyboard/keyboardeventfilter.h"
+#include "library/library.h"
+#include "library/relations/dlgrelationinfo.h"
+#include "library/relations/relationstablemodel.h"
+#include "moc_dlgrelations.cpp"
+#include "util/assert.h"
+#include "widget/wlibrary.h"
+#include "widget/wrelationtableview.h"
+
+DlgRelations::DlgRelations(
+        WLibrary* parent,
+        UserSettingsPointer pConfig,
+        Library* pLibrary,
+        KeyboardEventFilter* pKeyboard,
+        bool relationPairView)
+        : QWidget(parent),
+          Ui::DlgRelations(),
+          m_pRelationTableView(
+                  new WRelationTableView(
```

Comment:
We use our own parented_ptr and make_parented to assert that the widget has a parent and is not leaked.